cosi/README.md

```diff
@@ -78,6 +74,23 @@ cosi check -g $COTHORITY
 
 This will first contact each server individually and then check a few random collective signing group constellations. If there are connectivity problems, due to firewalls or bad connections, for example, you will see a "Timeout on signing" or similar error message.
 
+## Description
+The purpose of this work is to implement a robust and scalable consensus algorithm using CoSi protocol and handling some exceptions. **The CoSi tree is a three level tree** to make a compromise between the two-level tree, making the root-node vulnerable to DoS, and a more than three level tree, slowing the algorithm because of the RTT between the root node and the leaves.
+
+The tree is composed of a leader (root-node), and some groups of equal size, each having a sub-leader (second level nodes) and members (leaves). The groups composition are defined by the leader.
+
+We want to **handle non-responding nodes**, no matter where they are in the tree. If a leaf is failing, then it is ignored in the CoSi commitment. If a sub-leader is non-responding, then the leader (root node) recreates the group designing another sub-leader from the group members. And finally, if the leader is failing, the protocol restarts using another leader.
```

Comment:
s/designing/selecting/

Reply:
Is this "if leader is failing" part implemented already?

Reply:
No, will update the readme.